### ISSUE TITLE
Problem while saving model to s3 directory

### ISSUE THREAD

**Issue by Aditi00a** [JohnSnowLabs/spark-nlp#5900]
## Description
while saving the model to an s3 directory the model saves the metadata and fields inside the models folder however the tensorflow file is getting saved outside the folder with the same name as the model file path. 
this is creating a problem while loading the saved model

## Expected Behavior
the model directory should have the metadata, fields as well as he tensorflow files inside it on s3

## Current Behavior
while saving the model to an s3 directory the model saves the metadata and fields inside the models folder however the tensorflow file is getting saved outside the folder with the same name as the model file path

## Steps to Reproduce
1. train and save the model with
ner_model.stages[1].write().overwrite().save('./path_to_s3_dir')

## Context
unable to load the trained model from s3

## Your Environment

* Spark NLP version `sparknlp.version()`: 3.1.1
* Apache NLP version `spark.version`: 3.1.2
* Java version `java -version`:
* Setup and installation (Pypi, Conda, Maven, etc.):
* Operating System and version:
* Link to your project (if any):

**Comment by maziyarpanahi** [comment]
Hi,

could you please add more info regarding 

- the full code for saving and loading
- what is your platform, where are you running Spark NLP,
- a print/screenshot of what is saved inside `path_to_s3_dir` 
- and the full stack error

**Comment by Aditi00a** [comment]
**code used to train the model -** 
  args = parser.parse_args()

    spark = sparknlp.start()
    print("Spark NLP version: ", sparknlp.version())
    print("Apache Spark version: ", spark.version)

    training_file = "s3://" + os.path.join(args.s3_input_bucket,
                                           args.s3_input_key_prefix,
                                           args.training_file)

    testing_file = "s3://" + os.path.join(args.s3_input_bucket,
                                          args.s3_input_key_prefix,
                                          args.testing_file)


    training_data = CoNLL().readDataset(spark, training_file)
    training_data.show(10)

    # we use BERT Tiny
    bert_annotator = BertEmbeddings.pretrained('small_bert_L2_128', 'en'). \
        setInputCols(["sentence", 'token']). \
        setOutputCol("bert"). \
        setBatchSize(8)

    test_data = CoNLL().readDataset(spark, testing_file)
    test_data = bert_annotator.transform(test_data)

    test_data_parquet = "test.parquet"
    test_data.write.parquet(test_data_parquet)

    nerTagger = NerDLApproach(). \
        setInputCols(["sentence", "token", "bert"]). \
        setLabelColumn("label"). \
        setOutputCol("ner"). \
        setMaxEpochs(5). \
        setLr(0.001). \
        setPo(0.005). \
        setBatchSize(32). \
        setEvaluationLogExtended(True). \
        setEnableOutputLogs(True). \
        setTestDataset(test_data_parquet)

    pipeline = Pipeline(stages=[bert_annotator, nerTagger])

    ner_model = pipeline.fit(training_data)

    predictions = ner_model.transform(
        test_data.select("sentence", "token", "label"))

    eval_data = predictions.select(F.explode(
        F.arrays_zip('token.result', 'label.result', 'ner.result')).alias(
        "cols")) \
        .select(F.expr("cols['0']").alias("token"),
                F.expr("cols['1']").alias("ground_truth"),
                F.expr("cols['2']").alias("prediction"))

    eval_data.write.parquet("s3://" + os.path.join(args.s3_output_bucket,
                                                   args.s3_output_key_prefix,
                                                   "prediction"),
                            mode="overwrite"
                            )

    ner_model.stages[1].write().overwrite().save("s3://" +
                                                 os.path.join(
                                                     args.s3_output_bucket,
                                                     args.s3_output_key_prefix,
                                                     "NER_bert_2021_0730"))

**code used to predict the model ** 

    model_file = "s3://" + os.path.join(args.s3_output_bucket,
                                        args.s3_output_key_prefix,
                                        "NER_bert_2021_0730")

    testing_file = "s3://" + os.path.join(args.s3_input_bucket,
                                          args.s3_input_key_prefix,
                                          args.testing_file)

    document = DocumentAssembler() \
        .setInputCol("text") \
        .setOutputCol("document")

    sentence = SentenceDetector() \
        .setInputCols(['document']) \
        .setOutputCol('sentence')

    token = Tokenizer() \
        .setInputCols(['sentence']) \
        .setOutputCol('token')

    bert = BertEmbeddings.pretrained('small_bert_L2_128', 'en') \
        .setInputCols(["sentence", 'token']) \
        .setOutputCol("bert") \
        .setCaseSensitive(True)

    loaded_ner_model = NerDLModel.load(model_file) \
        .setInputCols(["sentence", "token", "bert"]) \
        .setOutputCol("ner")

    converter = NerConverter() \
        .setInputCols(["document", "token", "ner"]) \
        .setOutputCol("ner_span")

    ner_prediction_pipeline = Pipeline(
        stages=[
            document,
            sentence,
            token,
            bert,
            loaded_ner_model,
            converter])

    df = spark.read.csv(testing_file, header=True)

    empty_data = spark.createDataFrame([['']]).toDF("text")
    prediction_model = ner_prediction_pipeline.fit(empty_data)
    predictions = prediction_model.transform(df.withColumnRenamed("review",
                                                                  "text"))



**Running SparkNLP on Sagemaker docker**

**configuration -** 
configuration = [
    {
        "Classification": "spark-defaults",
        "Properties": {
            "spark.yarn.stagingDir": "hdfs:///tmp",
            "spark.yarn.preserve.staging.files": "true",
            "spark.kryoserializer.buffer.max": "2000M",
            "spark.serializer": "org.apache.spark.serializer.KryoSerializer",
            "spark.driver.maxResultSize": "0",
            "spark.jars.packages":
                "com.johnsnowlabs.nlp:spark-nlp-spark24_2.11:3.1.3",
            "spark.driver.memory": "12g",
            "spark.driver.cores": "3",
            "spark.executor.memory": "12g",
            "spark.executor.cores": "4",
            "spark.driver.memoryOverhead": "3g"
        }
    }
]

on s3 this is how the model files are saved. the 'tensorflow' file that is supposed to be inside the the model directory is outside with the same name as the directory('model')

![image](https://user-images.githubusercontent.com/88418260/128164933-a150f2ec-0822-45fc-9b27-d75d47d903b5.png)

![image](https://user-images.githubusercontent.com/88418260/128164989-271933c9-f9a4-47ec-b80f-e60269763fca.png)

during loading of the model, it is unable to find the tensorflow file within the directory.
![image](https://user-images.githubusercontent.com/88418260/128168548-2c38a72f-0d84-407a-a7eb-03d26c67faae.png)

**Comment by maziyarpanahi** [comment]
The `NER_bert_2021_0730` is not the model being outside. I don't know what that really is, but it's just a file with that name being saved. The directory of `NER_bert_2021_0730/` however is missing a file called `tensorflow` next to fields and metadata directories. (this `tensorflow` file cannot be renamed by you or anyone so it's just not there).

Could you please just simply following these steps:

- remove everything that already exists, inside, outside, all of it
- simple use the following code without any `arg` or dynamic variables, just a hardcoded value so we know what's what:

```python
spark = sparknlp.start()
print("Spark NLP version: ", sparknlp.version())
print("Apache Spark version: ", spark.version)

training_file = "s3://" + os.path.join(args.s3_input_bucket,
                                       args.s3_input_key_prefix,
                                       args.training_file)

testing_file = "s3://" + os.path.join(args.s3_input_bucket,
                                      args.s3_input_key_prefix,
                                      args.testing_file)


training_data = CoNLL().readDataset(spark, training_file)
training_data.show(10)

# we use BERT Tiny
bert_annotator = BertEmbeddings.pretrained('small_bert_L2_128', 'en'). \
    setInputCols(["sentence", 'token']). \
    setOutputCol("bert"). \
    setBatchSize(8)

test_data = CoNLL().readDataset(spark, testing_file)
test_data = bert_annotator.transform(test_data)

test_data_parquet = "test.parquet"
test_data.write.parquet(test_data_parquet)

nerTagger = NerDLApproach(). \
    setInputCols(["sentence", "token", "bert"]). \
    setLabelColumn("label"). \
    setOutputCol("ner"). \
    setMaxEpochs(5). \
    setLr(0.001). \
    setPo(0.005). \
    setBatchSize(32). \
    setEvaluationLogExtended(True). \
    setEnableOutputLogs(True). \
    setTestDataset(test_data_parquet)

pipeline = Pipeline(stages=[bert_annotator, nerTagger])

ner_model = pipeline.fit(training_data)

# you can set the path to anything, but make sure nothing exists there to be sure what happened
ner_model.stages[1].write().overwrite().save("s3://NER_small_bert_L2_128_model"))
```

That's it, let's just see with hard-coded S3 path what we see when it's saved because I just tried and since `.save()` is a Spark ML feature it correctly saves everything in that path - (also pay attention to any error and logs, it may have failed to actually save the file)

**Comment by Aditi00a** [comment]
so when i moved "NER_bert_2021_0730" file inside the folder and remaned it to "tensorflow" 
then loaded the model then it  worked. 
this confirms that that file is the model

**Comment by maziyarpanahi** [comment]
That could be, but let's follow that code in a clean directory to be 100% sure of what happened really since that file cannot be named/renamed. (it's hardcoded inside the code). Please follow the code with a simple hard-coded S3 path in a place that nothing exists so we can see if this happens again.
